jetty-util/src/test/java/org/eclipse/jetty/util/resource/PathResourceTest.java

```diff
@@ -113,4 +114,16 @@ public void testDefaultFileSystemGetFile() throws Exception
         File file = resource.getFile();
         assertThat("File for default FileSystem", file, is(exampleJar.toFile()));
     }
+
+    @Test
+    public void testSame() throws Exception
```

Comment:
I would add an assertion that checks that symlink'ed paths behave as they should as this is the original root cause.

Something like this (except that I do not know if that would work on windows):

```java

Path epath2 = Files.createSymbolicLink(MavenTestingUtils.getTargetPath().resolve("testSame-symlink"), epath.getParent()).resolve("example.jar");
PathResource ePathResource2 = new PathResource(epath2);

assertThat(ePathResource.isSame(ePathResource2), Matchers.is(true));
assertThat(ePathResource.equals(ePathResource2), Matchers.is(false));
```